lib/python/ensembl/brc4/runnable/amend_genome_metadata.py

```diff
@@ -0,0 +1,198 @@
+            return []
+        
+        seq_regions = []
+        _open = gbff_path.endswith(".gz") and gzip.open or open
+        with _open(gbff_path, 'rt') as gbff_file:
+
+            for record in SeqIO.parse(gbff_file, "genbank"):
+                seq_regions.append(record.id)
+                    
+        return seq_regions
+    
+    def report_to_csv(self, report_path: str) -> Tuple[str, dict]:
+        """Load an assembly report as a csv string.
+
+        Args:
+            report_path: path to a seq_region file from INSDC/RefSeq
+        
+        Returns:
+            The data as a string in csv format, and the head metadata as a dict.
+        """
+
+        _open = report_path.endswith(".gz") and gzip.open or open
+        with _open(report_path, 'rt') as report:
+            data = ""
+            metadata = {}
+            last_head = ""
+            for line in report:
+                # Ignore header
+                if line.startswith("#"):
+                    # Get metadata values if possible
+                    match = re.search("# (.+?): (.+?)$", line)
+                    if match:
+                        metadata[match.group(1)] = match.group(2)
+                    last_head = line
+                    continue
+                else:
+                    if last_head:
+                        data += last_head[2:].strip() + "\n"
+                        last_head = ""
+                    data += line
+            return data, metadata
+    
+    def get_report_regions(self, report_path: str) -> List[str]:
+        """Get seq_region data from report file.
+
+        Args:
+            report_path: Path to the seq_regions report from INSDC/RefSeq.
+            use_refseq: True if using RefSeq, False if INSDC.
+        
+        Returns:
+            A list of SeqRegion names.
+        """
+
+        # Get the report in a CSV format, easier to manipulate
+        report_csv, _ = self.report_to_csv(report_path)
+        
+        # Feed the csv string to the CSV reader
+        reader = csv.DictReader(report_csv.splitlines(), delimiter="\t", quoting=csv.QUOTE_NONE)
+        
+        # Create the seq_regions
+        seq_regions = []
+        for row in reader:
+            refseq_name = row["RefSeq-Accn"]
+            genbank_name = row["GenBank-Accn"]
+            name = ''
+            if genbank_name and genbank_name != "na":
+                name = genbank_name
+            elif refseq_name and refseq_name != "na":
+                name = refseq_name
+            if name:
+                seq_regions.append(name)
+        
+        return seq_regions
+
+    def print_json(self, path: str, data: Any) -> None:
+        """Generic data json dumper to a file.
+        
+        Args:
+            path: Path to the json to create.
+            data: Any data to store.
+         """
+        with open(path, "w") as json_out:
+            json_out.write(json.dumps(data, sort_keys=True, indent=4))
```

Comment:
In this case, since the function is only called once, I would suggest to replace its call for these two lines, unless you plan to extend this runnable in the future and this method will be used more.

Reply:
I've put that function (as well as get_json) in a Utils module, since they are used multiple times by other runnables as well.